src/lib.rs

```diff
@@ -156,21 +157,17 @@ pub fn select_coin_fifo(
         {
             break;
         }
-        accumulated_value += i.value;
-        accumulated_weight += i.weight;
-        if let Some(index) = inputs.iter().position(|x| x == i) {
-            selected_inputs.push(index);
-        } else {
-            break;
-        }
+        accumulated_value += inputs.value;
+        accumulated_weight += inputs.weight;
+        selected_inputs.push(index);
     }
     if accumulated_value < options.target_value + estimated_fees + options.min_drain_value {
```

Comment:
There could be cases where estimated_fees < min_absolute_fee, refer SRD to fix this.

Reply:
Right.